### ISSUE TITLE
PPO1 and Distributed Training

### ISSUE THREAD

**Issue by luckeciano** [openai/baselines#526]
Hello guys,

I'm using the PPO1 algorithm to execute a distributed training in a specific environment that we've integrated previously. My question is more about how it works.

As far as I understood, the distributed training happens synchronously, averaging the gradients calculated from each worker. So, it seems to me that if I we have 50 workers initially and then change for 100 workers, we will get almost twice of data, but keep doing the same number of updates - which seems to me I'm losing data efficiency. I tried to get more updates faster (make smaller batches and minibatches, or even more optimization epochs, which made the kl loss diverge), but the algorithm converges worse than when I use a single worker. 

Can anyone explain me how the distributed training works? and if I really need to change the hyperparameters like this to achieve better learning or ideally the same hyperparameters should work with a single worker and multiple workers?

Thanks in advance!

**Comment by pzhokhov** [comment]
Hi @luckeciano! Your understanding is correct - the number of updates does not change in the scenario you are describing, rather the batch size to compute update is increasing providing more accurate updates. As for the hyperparameters that work in the distributed setting (especially with 50-100 workers) - I am afraid I don't have a good answer to that, I don't think we have experimented much in that direction (pinging @joschu, he may have a better answer to this). If you need faster convergence in the distributed setting, I'd recommend trying larger learning rates out of general sense that with bigger batches your gradients are more accurate, so it is safer to take bigger steps in the direction of the gradients.
